tools/explorer/tt_adapter/src/tt_adapter/main.py

```diff
@@ -43,11 +43,19 @@ def __init__(self):
     def convert(
         self, model_path: str, settings: Dict
     ) -> model_explorer.ModelExplorerGraphs:
+        perf_trace = None
+        if optimized_model_path := self.model_runner.get_optimized_model_path():
+            print(f"Using optimized model: {optimized_model_path}")
+            model_path = optimized_model_path
+
+            # Get performance results.
+            perf_trace = self.model_runner.get_perf_trace()
+
         module = utils.parse_mlir_file(model_path)
 
         # Convert TTIR to Model Explorer Graphs and Display/Return
-        graph = mlir.build_graph(module)
-        return {"graphs": [graph]}
+        graph, perf_data = mlir.build_graph(module, perf_trace)
+        return {"graphs": [graph], "perf_data": perf_data}
```

Comment:
The `perf_data` will not be returned because it needs to be wrapped into an adapter_response and kept in `graphs`. It should look something like:

```
return to_adapter_format({
  "graphs": [graph],
  "perf_data": perf_data
})
```

In case the graph isn't passed we should escalate this to change how the frontend parses this as well since the graphs: graph property needs to exist.